erpnext/stock/doctype/serial_no/serial_no.py

```diff
@@ -431,7 +431,6 @@ def update_args_for_serial_no(serial_no_doc, serial_no, args, is_new=False):
 		"via_stock_ledger": args.get("via_stock_ledger") or True,
 		"supplier": args.get("supplier"),
 		"location": args.get("location"),
-		"sales_order": frappe.db.get_value(args.get("voucher_type"), args.get("voucher_no"), "sales_order_no"),
```

Comment:
@meshramsaroj Yes this line needs to be delete. but you have to fix the functionality why it was added here.
First of all, check for on_submit method in stock entry.
update_so_in_serial_number this function update sales order in serial number.

workflow: In Sales Order Item there is a checkbox ensure_delivery_based_on_produced_serial_no is set then it will automatically update the Sales order in the serial number.

DOD:
In Sales Order Item If the item is serialized item then ensure_delivery_based_on_produced_serial_no should be check by default.

Reply:
https://github.com/Bloomstack/erpnext/pull/1742 Once Merge this. We can close this